packages/core/src/utils/ConfigurationLoader.ts

```diff
@@ -17,7 +17,11 @@ export class ConfigurationLoader {
 
       if (await pathExists(path)) {
         // eslint-disable-next-line @typescript-eslint/no-var-requires
-        const config = require(path);
+        const configImport = require(path);
+        let config = configImport.default || configImport;
+        if (config.then && config.catch) {
+          config = await config;
+        }
         return new Configuration({ ...(config.default || config), ...options }, validate);
```

Comment:
let's do it this way:

```suggestion
        const configImport = require(path);
        let config = configImport.default || configImport;

        if (config instanceof Promise) {
          config = await config;
        }

        return new Configuration({ ...config, ...options }, validate);
```

Reply:
actually, it can be even simpler :-), like:

`const config = require(path);
return new Configuration({ ...(await (config.default || config)), ...options }, validate);`

I've updated the docs